src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ConstructorParametersShouldMatchPropertyAndFieldNames.cs

```diff
@@ -0,0 +1,261 @@
+
+                if (referencedSymbol == null || referencedSymbol.IsStatic)
+                {
+                    return;
+                }
+
+                IParameterSymbol param = operation.Parameter;
+                var field = referencedSymbol as IFieldSymbol;
+                var prop = referencedSymbol as IPropertySymbol;
+
+                if (field == null && prop == null)
+                {
+                    return;
+                }
+
+                if (IsSupportedField(field) && !IsParamMatchFieldName(param, field))
+                {
+                    var properties = ImmutableDictionary<string, string?>.Empty.SetItem(ReferencedFieldOrPropertyName, field.Name);
+
+                    context.ReportDiagnostic(
+                        param.CreateDiagnostic(
+                            FieldRule,
+                            properties,
+                            param.ContainingType.ToDisplayString(SymbolDisplayFormats.ShortSymbolDisplayFormat),
+                            param.Name,
+                            field.Name));
+
+                    return;
+                }
+
+                if (IsSupportedProp(prop))
+                {
+                    if (!IsParamMatchPropName(param, prop))
+                    {
+                        ReportPropertyDiagnostic(context, PropertyNameRule, ParameterDiagnosticReason.NameMismatch, param, prop);
+                    }
+
+                    if (!prop.IsPublic())
+                    {
+                        ReportPropertyDiagnostic(context, PropertyPublicRule, ParameterDiagnosticReason.PropertyInappropriateVisibility, param, prop);
+                    }
+                }
+            }
+
+            private static bool IsSupportedProp([NotNullWhen(true)] IPropertySymbol? prop)
+            {
+                if (prop == null)
+                {
+                    return false;
+                }
+
+                return true;
+            }
+
+            private static bool IsSupportedField([NotNullWhen(true)] IFieldSymbol? field)
+            {
+                if (field == null)
+                {
+                    return false;
+                }
+
+                return true;
+            }
+
+            private static bool IsParamMatchFieldName(IParameterSymbol param, IFieldSymbol field)
+            {
+                if (param.Name.Length != field.Name.Length)
+                {
+                    return false;
+                }
+
+                var paramWords = WordParser.Parse(param.Name, WordParserOptions.SplitCompoundWords);
+                var fieldWords = WordParser.Parse(field.Name, WordParserOptions.SplitCompoundWords).ToImmutableArray();
+
+                return paramWords.All(x => WordParser.ContainsWord(x, WordParserOptions.SplitCompoundWords, fieldWords));
+            }
+
+            private static bool IsParamMatchPropName(IParameterSymbol param, IPropertySymbol prop)
```

Comment:
This method is duplicate of the above.

Consider changing `IPropertySymbol prop` to `ISymbol fieldOrProperty` and see if it works.

Reply:
I had introduced two separate methods because props and fields can use different naming styles. But JsonConstructor doesn't respect naming styles, so it can be merged into a single method. Done.